modules/grpc/bigquery/bigquery-worker.cpp

```diff
@@ -311,9 +319,16 @@ DestinationWorker::insert(LogMessage *msg)
   this->batch_size++;
   rows->add_serialized_rows(message->SerializePartialAsString());
 
+  row_bytes = rows->serialized_rows(rows->serialized_rows_size() - 1).size();
```

Comment:
Is this one taking the size of the string we just added a line up? Would it be more readable if we first took the reference of the serialized message locally and the take it's size and also add it using add_serialized_rows()

Just a nitpick. I had some questions here a serialized_rows doesn't look like an array I have to assume that rows->serialized_rows(rows->serialized_rows_size() - 1) returns the last message.

Would be better to see that explicitly

Reply:
I might be wrong, but I am afraid that if we store `message->SerializePartialAsString()` in a variable, the `rows->add_serialized_rows()` will not move the `std::string`, but copy it.

Reply:
Some readability is not worth that price. But maybe it would be worthy of a comment

Reply:
```
message ProtoRows {
  // A sequence of rows serialized as a Protocol Buffer.
  //
  // See https://developers.google.com/protocol-buffers/docs/overview for more
  // information on deserializing this field.
  repeated bytes serialized_rows = 1;
}
```

`serialized_rows` is a repeated value, which we should be able to access with the generated `rows->serialized_rows(i)` function.

See: https://protobuf.dev/getting-started/cpptutorial/#protobuf-api

> Repeated fields also have some special methods – if you look at the methods for the repeated phones field, you’ll see that you can check the repeated field’s _size (in other words, how many phone numbers are associated with this Person).
> * get a specified phone number using its index.
> * update an existing phone number at the specified index.
> * add another phone number to the message which you can then edit (repeated scalar types have an add_ that just lets you pass in the new value).

Reply:
My IDE tells me that it uses copy assignment when we store it in a variable, in contrary to the move assignment when we do not store it:

---

![Screenshot from 2023-12-15 13-31-12](https://github.com/syslog-ng/syslog-ng/assets/45236571/e15bd8aa-0be1-4d0d-91c1-cc970a142130)

---

![Screenshot from 2023-12-15 13-31-26](https://github.com/syslog-ng/syslog-ng/assets/45236571/99869bdd-2202-4051-aebd-e6a5d41afd91)